server/storage/mvcc/kvstore_compaction.go

```diff
@@ -63,6 +64,7 @@ func (s *store) scheduleCompaction(compactMainRev, prevCompactRev int64) (KeyVal
 			// gofail: var compactBeforeSetFinishedCompact struct{}
 			UnsafeSetFinishedCompact(tx, compactMainRev)
 			tx.Unlock()
+			dbCompactionPauseMs.Observe(float64(time.Since(start) / time.Millisecond))
```

Comment:
This metric was introduced in 2015 - https://github.com/etcd-io/etcd/commit/b5838edb93.
And in this patch - https://github.com/etcd-io/etcd/pull/11034, the write buffer is explicitly forced into bbolt, except for the last round, which skips the flush and allows the backend to commit the buffer in the background. However, LockOutsideApply() still holds the lock. Therefore, I believe we should record the pause duration for the last round, even if it doesn't flush any data.